### PR TITLE
VZ-7276: Add simple make precommit targets to be run locally

### PR DESCRIPTION
**Pull request by bobdonat**
This pull request adds precommit make targets that can be run locally before pushing your branch to be run with Jenkins.  The idea is to be able to catch errors locally and sooner instead of waiting for the errors to be caught in a Jenkins job.

The precommit targets can run all the linters, copyright check, build all go code, and run all unit-tests.

Some refactoring was done with tests to get things to work.

### CODE REVIEW

File: tools/eventually-checker/testdata/go.mod [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) 2020, 2022, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+module eventually-checker
+
+go 1.17
```
Comment (anders-swanson) [inline tools/eventually-checker/testdata/go.mod:6]: go 1.19?

File: Makefile [hunk 2]
```diff
@@ -69,6 +69,27 @@ test-platform-operator-remove:
 test-platform-operator-install-logs:
 	kubectl logs -f -n default $(shell kubectl get pods -n default --no-headers | grep "^verrazzano-install-" | cut -d ' ' -f 1)
 
+.PHONY: precommit
+precommit: precommit-check precommit-build unit-test-coverage
```
Comment (anders-swanson) [inline Makefile:73]: Should the jenkinsfile be calling these?

Comment (bobdonat) [reply Makefile:73]: yes, that is the thinking. @gilbode wanted to get this in as a start

File: tools/eventually-checker/testdata/go.mod (hunk 1 above)
Comment (bobdonat) [reply tools/eventually-checker/testdata/go.mod:6]: removed this file. not sure it was there.

Comment (gilbode) [reply tools/eventually-checker/testdata/go.mod:6]: make sure eventually checker unit tests still pass without it

**Review by anders-swanson — APPROVED** [review]

File: tools/eventually-checker/testdata/go.mod (hunk 1 above)
Comment (bobdonat) [reply tools/eventually-checker/testdata/go.mod]: yes, unit tests are passing without it

**Pull request merged**